### PR TITLE
Display a notification when the language server crashes

### LINKED ISSUE #409: No error is reported to the user when the binary fails to start
I installed [0.15.0-108](https://download.jboss.org/jbosstools/vscode-xml/staging/vscode-xml-0.15.0-108.vsix) on my Mac, set preferBinary=true, and after downloading the binary, the server fails to start but no error message pops up to the user.

You need to open the XML output view to see:
```
[Error - 9:17:15 AM] Connection to server is erroring. Shutting down server.
[Error - 9:17:15 AM] Connection to server is erroring. Shutting down server.
[Error - 9:17:15 AM] Connection to server is erroring. Shutting down server.
```

### PR DESCRIPTION
**Pull request by datho7561**
Closes #409

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
To test (under Linux), I made a file `lemminx-linux` under the `server` directory that contained: `exit`, then made it executable. I then set `xml.server.preferBinary = true`, and ran the extension. A notification about the server crashing should show up.

File: src/extension.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import * as os from 'os';
 import * as path from 'path';
 import { Command, commands, ExtensionContext, extensions, IndentAction, LanguageConfiguration, languages, Position, TextDocument, TextEditor, Uri, window, workspace } from "vscode";
-import { CancellationToken, ConfigurationParams, ConfigurationRequest, DidChangeConfigurationNotification, Executable, ExecuteCommandParams, ExecuteCommandRequest, LanguageClient, LanguageClientOptions, MessageType, NotificationType, ReferencesRequest, RequestType, RevealOutputChannelOn, TextDocumentIdentifier, TextDocumentPositionParams } from 'vscode-languageclient';
+import { CancellationToken, ConfigurationParams, ConfigurationRequest, DidChangeConfigurationNotification, Executable, ExecuteCommandParams, ExecuteCommandRequest, LanguageClient, LanguageClientOptions, MessageType, NotificationType, ReferencesRequest, RequestType, RevealOutputChannelOn, TextDocumentIdentifier, TextDocumentPositionParams, InitializationFailedHandler, FoldingRangeRequest, ErrorAction, CloseAction } from 'vscode-languageclient';
```
Comment (fbricon) [inline src/extension.ts:16]: do you need the InitializationFailedHandler, FoldingRangeRequest imports?

Comment (datho7561) [reply src/extension.ts]: Good catch! I removed the imports.

**Review by rgrunber — COMMENTED** [review]
This works for me, so I think once conflicts are resolved, we should merge. One thing to keep in mind is that in vscode-java the restart behaviour tries to restart a few times after a crash and only stays down when it looks like the crashes are happening too often.

https://github.com/redhat-developer/vscode-java/blob/master/src/extension.ts#L54

Something to maybe consider in the future.

**Comment by datho7561** [comment]
Okay. I resolved the merge conflicts. I agree that we should support attempting to relaunch the server like in vscode-java

**Comment by datho7561** [comment]
I switched over to the method that vscode-java (stop trying to restart the server if the server has crashed 5 times in the last 3 minutes)

**Review by rgrunber — APPROVED** [review]
Feel free to merge this when ready. Tried it out and the crash timer seems to be working well.

**Pull request merged**